apps/signing/views.py

```diff
@@ -13,24 +14,29 @@
 from signing.serializers import FileUploadSerializer
 
 
-def with_addon(fn):
-    def inner(view, request, guid, version):
-        try:
-            addon = Addon.unfiltered.get(guid=guid)
-        except Addon.DoesNotExist:
-            return Response({'error': _('Could not find addon.')},
-                            status=status.HTTP_404_NOT_FOUND)
-        if not addon.has_author(request.user):
-            return Response(
-                {'error': _('You do not own this addon.')},
-                status=status.HTTP_403_FORBIDDEN)
-        return fn(view, request, addon, version)
-    return inner
+def with_addon(allow_missing=False):
```

Comment:
because this has a couple different modes now, a quick docstring would be helpful